// monaco-editor/esm/vs/basic-languages/xml/xml.contribution.js
define('node_modules/monaco-editor/esm/vs/basic-languages/xml/xml.contribution', function(require, exports, module) {

  /*---------------------------------------------------------------------------------------------
   *  Copyright (c) Microsoft Corporation. All rights reserved.
   *  Licensed under the MIT License. See License.txt in the project root for license information.
   *--------------------------------------------------------------------------------------------*/
  'use strict';
  var __importStar = (this && this.__importStar) || function (mod) {
      if (mod && mod.__esModule) return mod;
      var result = {};
      if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
      result["default"] = mod;
      return result;
  };
  Object.defineProperty(exports, "__esModule", { value: true });
  var __contribution_js_1 = require("node_modules/monaco-editor/esm/vs/basic-languages/_.contribution");
  __contribution_js_1.registerLanguage({
      id: 'xml',
      extensions: ['.xml', '.dtd', '.ascx', '.csproj', '.config', '.wxi', '.wxl', '.wxs', '.xaml', '.svg', '.svgz'],
      firstLine: '(\\<\\?xml.*)|(\\<svg)|(\\<\\!doctype\\s+svg)',
      aliases: ['XML', 'xml'],
      mimetypes: ['text/xml', 'application/xml', 'application/xaml+xml', 'application/xml-dtd'],
      loader: function () { return Promise.resolve().then(function () { return __importStar(require('node_modules/monaco-editor/esm/vs/basic-languages/xml/xml')); }); }
  });
  

});
